File: src/main/java/com/hank/guess/Game.java
package com.hank.guess;


import java.util.Random;
import java.util.Scanner;

public class Game {
    public static void main(String[] args) {
        Random random = new Random();
        int secret = random.nextInt(10)+1; //bound:10產生0-9，故由1開始要+1
        System.out.println(secret);
        Scanner scanner = new Scanner(System.in);
        int number = 0;
        while (number != secret) {
            System.out.print("Please enter a number: ");
            number = scanner.nextInt();
            if (number < secret) {
                System.out.println("Higher");
            } else if (number > secret) {
                System.out.println("Lower");
            } else {
                System.out.println("Great , the number is " + number);
            }
        }

//        do while練習
        /*int i = 0;
        do {
            System.out.println(i);
            i++; //i++ => i=i+1
        } while(i < 3);*/

    }
}

File: src/main/kotlin/com/hank/Game.kt
package com.hank


import java.util.Random
import java.util.Scanner

fun main() {
    val secret = Random().nextInt(10) + 1
    println(secret)
//    val scanner = Scanner(System.`in`) //in是kotlin關鍵字，要加鍵盤1左邊「`」符號
    var number = 0
    while (number != secret) {
        print("Please enter a number : ")
//        number = scanner.nextInt()
        number = readln()!!.toInt() //kotlin自已的Scanner，!!表示自行處理例外 exception問題
        if (number > secret) {
            println("Lower")
        } else if (number < secret) {
            println("Higher")
        } else {
            println("Great, the number is $number")
        }

    }
}